Autopilot/AttitudeManager/AttitudeManager.c

```diff
@@ -920,9 +919,11 @@ void readDatalink(void){
                 setKValues(GAIN_KI,(float*)(&cmd->data));
                 break;
             case SET_GAINS:
-                int type = (int*) (&cmd->data);
-                setGains(type,(float*)(&cmd->data));
+            {
+                char* channel = (char*) (&cmd->data);
+                setGains(*channel,(float*)(&cmd->data));
```

Comment:
I believe this is now functionally correct, however it is only correct because the method calls with a built in index offset. This means of you ever call from another location, it is likely to fail in exciting ways. I recommend doing the offset correction on this line and zero indexing in `setGains`. Alternatively, you could make a dedicated struct that describes the command format, and change the methods signatures as appropriate.